modules/client/src/addressList/internal/client/methods.ts

```diff
@@ -130,6 +122,22 @@ export class ClientAddressListMethods extends ClientCore
       proposalId: hexZeroPad(parsedLog.args["voteId"].toHexString(), 32),
     };
   }
+
+  /**
+   * Pins a metadata object into IPFS and retruns the generated hash
+   *
+   * @param {ProposalMetadata} params
+   * @return {*}  {Promise<string>}
+   * @memberof ClientMethods
+   */
+  public async pinMetadata(params: ProposalMetadata): Promise<string> {
```

Comment:
this function doesn't pin the data on the cluster. but rather just adds it. The ipfs cluster will delete this files with the next GC run. 
So I suggest we either actually pin the data in this function or rename it to `addMetadata` for example

Reply:
maybe it is also worth putting it into one of the common folders/packages to remove code duplication. It is the same function everywhere

Reply:
Well, then I would add a call to `this.ipfs.pin()` instead of renaming this to `addMetadata()` and letting GC dispose it?

Reply:
I duplicated it mainly for the return of the function, being `IMetadata` or `ProposalMetadata`. Between the two voting plugin the function is exacly the same so I'll move it to the common. 

About the pin thin I tried to edit the `add` function in `core` to add and then pin the result and rename the function to `addAndPin`, but it kept throwing errors when trying to `pin`, I'm not entirely sure why

Reply:
then I suggest you send me the error messages you get so I can fix it for you. Otherwise the naming is wrong

Reply:
I strongly recommend having the `add` and `pin` functions as their own thing... and not `addAndPin`